src/Tools.js

```diff
@@ -2,6 +2,33 @@
  * Class to hold some global helper tools
  */
 class Tools {
+	/**
+	 * Output into the console.
+	 *
+	 * @param  {string} [type='info'] Log type used for console output
+	 *  valid options are:
+	 *  - 'info' for console.info()
+	 *  - 'warn' for console.warn()
+	 *  - 'error' for console.error()
+	 * @param  {string} str String to output
+	 * @param {Mixed} [params] Parameters to output into the console
+	 *  alongside the message. This is especially useful for errors.
+	 */
+	static log( type = 'info', str = '', params ) {
+		const prefix = 'Who Wrote That extension: ';
+		let method = window.console.info;
+		if ( type === 'warn' ) {
+			method = window.console.warn;
+		} else if ( type === 'error' ) {
+			method = window.console.error;
+		}
+
+		if ( params ) {
+			method( prefix + str, params );
+		} else {
+			method( prefix + str );
+		}
```

Comment:
You might could simplify lines 19-30 with `window.console[type]( prefix + str, params )`, setting an empty string as the default value for `params` on line 17. The idea here is that `log( 'foo', 'whatever' )` is the same as `console.whatever( 'foo' )`, whereby we trust we use something valid for the type.

Reply:
Ooh! Good point!